services/gateway/gateway-service.mjs

```diff
@@ -185,15 +185,22 @@ class GatewayService {
     }
 
     // Synchronize the state.
-    let firstLocation = {playerID: playerID, 
+    const firstLocation = {playerID: playerID,
       displayName: socket.playerData.displayName};
     firstLocation.mapCoord = socket.playerData.mapCoord;
+    let setLocation = false;
+    if (firstLocation.mapCoord === undefined) {
```

Comment:
I would strongly recommend always running the spawn point assignment. To spawning existing user into where they previously left, we'll need to occupy the cell, and if that fails we'll need more algorithms. Thus, for now we should default to assigning spawn points for all new connections.

Reply:
I agree with this. I suggest adding a TODO to remind you of finishing this feature in the future.